src/sentry/testutils/cases.py

```diff
@@ -1232,6 +1235,106 @@ def _send_buckets(cls, buckets, entity):
         )
 
 
+class BaseMetricsLayerTestCase(ABC, BaseMetricsTestCase):
+    @abstractmethod
+    def now(self):
+        """
+        Returns the current time instance that will be used throughout the tests of the metrics layer.
+
+        This method has to be implemented in all the children classes because it serves as a way to standardize
+        access to time.
+        """
+        raise NotImplementedError
+
+    def _store_metric_in_metrics_layer(
+        self,
+        type,
+        name,
+        tags,
+        value,
+        use_case_id,
+        org_id=None,
+        project_id=None,
+        hours_before_now=0,
+        minutes_before_now=0,
+        seconds_before_now=0,
+    ):
+        # We subtract one second in order to account for right non-inclusivity in the query. If we wouldn't do this
+        # some data won't be returned (this applies only if we use self.now() in the "end" bound of the query).
+        #
+        # Use SENTRY_SNUBA_INFO=true while running queries in tests to know more about how data is actually queried
+        # at the clickhouse level.
+        #
+        # The solution proposed aims at solving the problem of flaky tests that occurred during CI at specific times.
+        self.store_metric(
+            org_id=self.organization.id if org_id is None else org_id,
+            project_id=self.project.id if project_id is None else project_id,
+            type=type,
+            name=name,
+            tags=tags,
+            timestamp=(
+                self.now()
+                - timedelta(
+                    hours=hours_before_now,
+                    minutes=minutes_before_now,
+                    seconds=seconds_before_now + 1,
```

Comment:
Can you explain the +1  on the seconds to me ?

Reply:
Yes, we discussed this together in the meeting, we subtract -1 to account for right non-inclusivity, in this case +1 is because we do time - (+1) seconds.